doc: Doc check and fixes for 2.7

umapraseeda opened the pull request:
NCSDK-27762 : Doc check and fixes for 2.7
Link check + toc check

rugeGerritsen reviewed (approved):
Softdevice controller doc changes are good.

seko-nordic commented:
> Softdevice controller doc changes are good.

I would like to change following bullet:
> * The deprecated functions :c:func:`rand_prio_low_get` and :c:func:`rand_prio_high_get` have been removed.
>   This change does not affect applications developed in the |NCS| context. (DRGN-20473)
to
```
* The deprecated  :c:member:`sdc_rand_source_t.rand_prio_low_get` and :c:member:`sdc_rand_source_t.rand_prio_high_get` have been removed.
  This change does not affect applications developed in the |NCS| context. (DRGN-20473)
```
Otherwise it looks like we removed some APIs provided by SDC. In fact, we removed APIs that were used by SDC, so application needs to provide less now.

umapraseeda commented:
> > Softdevice controller doc changes are good.
> 
> I would like to change following bullet here
> 
> https://github.com/nrfconnect/sdk-nrfxlib/blob/0c1d011a1caae8771a764ab72a930903b35fae53/softdevice_controller/CHANGELOG.rst?plain=1#L38
> 
> > * The deprecated functions :c:func:`rand_prio_low_get` and :c:func:`rand_prio_high_get` have been removed.
> 
> to
> 
> ```
> * The deprecated  :c:member:`sdc_rand_source_t.rand_prio_low_get` and :c:member:`sdc_rand_source_t.rand_prio_high_get` have been removed.
> ```
> 
> Otherwise it looks like we removed some APIs provided by SDC. In fact, we removed APIs that were used by SDC, so application needs to provide less now.

Fixed

seko-nordic commented:
> Fixed

Thank you, looks good

peknis commented on softdevice_controller/CHANGELOG.rst:
```diff
@@ -35,7 +35,7 @@ Changes
   In cases where there are many conflicting activities within the scan window, this change may result in reduced scanning time. (DRGN-19050)
 * Scan windows are no longer limited to 16 seconds. (DRGN-19050)
-* The deprecated functions :c:func:`rand_prio_low_get` and :c:func:`rand_prio_high_get` have been removed.
+* The deprecated  :c:member:`sdc_rand_source_t.rand_prio_low_get` and :c:member:`sdc_rand_source_t.rand_prio_high_get` have been removed.
```
```suggestion
* The deprecated :c:member:`sdc_rand_source_t.rand_prio_low_get` and :c:member:`sdc_rand_source_t.rand_prio_high_get` have been removed.
```

peknis replied:
Now that these structure members are removed, should we just have them between double backticks instead of the current formatting?
